### PR TITLE
Fix several concurrency issue of analytics

### PR DESCRIPTION
**Pull request by duckofyork**

### CODE REVIEW

**Comment by duckofyork** [comment]
please retest this please

File: grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/SumQueryImpl.java [hunk 1]
```diff
@@ -45,14 +48,16 @@
         long startTime = System.currentTimeMillis();
 
         initSubGraph();
-        String dataType = checkSelectedResourceTypesHaveCorrectDataType(statisticsResourceTypeLabels);
+        ResourceType.DataType dataType = getDataTypeOfSelectedResourceTypes(statisticsResourceTypeLabels);
         if (!selectedResourceTypesHaveInstance(statisticsResourceTypeLabels)) return Optional.empty();
         Set<TypeId> allSubTypeIds = convertLabelsToIds(getCombinedSubTypes());
         Set<TypeId> statisticsResourceTypeIds = convertLabelsToIds(statisticsResourceTypeLabels);
 
+        String randomId = Integer.toString(ThreadLocalRandom.current().nextInt(Integer.MAX_VALUE));
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/SumQueryImpl.java]: It's super boring, but we should have a shared static method to do this.

Comment (duckofyork) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/SumQueryImpl.java]: I agree!

**Comment by fppt** [comment]
Merged into master here: https://github.com/graknlabs/grakn/pull/1653

**Pull request merged**